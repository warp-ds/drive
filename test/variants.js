import { describe, expect, test } from 'vitest';

import { setup } from './_helpers.js';

setup();

describe('variants', () => {
  test('negative', async ({ uno }) => {
    const classes = ['-inset-16', '-mt-32', '-bottom-1/2'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('breakpoints', async ({ uno }) => {
    const classes = ['md:mt-32', 'at-md:mt-32', 'lt-md:mt-32'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('important', async ({ uno }) => {
    const classes = ['!mb-32', '!sm:mt-32', '!md:-bottom-1/2'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('pseudo', async ({ uno }) => {
    const classes = [
      'last:mb-0',
      'before:p-32',
      'focus:border',
      'group',
      'group-hover:p-16',
      'group/llama',
      'group-hover/llama:m-32',
      'part-[part-name]:s-bg-primary',
    ];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('custom', async ({ uno }) => {
    const classes = ['last-child:mb-0'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });

  test('data variant', async ({ uno }) => {
    const classes = ['data-[is-active]:s-bg-primary'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });

  test('aria variant', async ({ uno }) => {
    const classes = ['aria-[invalid=spelling]:s-bg-primary'];
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  // space/divide variant is already tested by the space/divide classes
});
