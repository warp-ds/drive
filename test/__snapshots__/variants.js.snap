// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`variants > aria variant 1`] = `
"/* layer: default */
.aria-\\[invalid\\=spelling\\]\\:s-bg-primary[aria-invalid=spelling]{background-color:var(--w-s-color-background-primary);}"
`;

exports[`variants > breakpoints 1`] = `
"/* layer: default */
@media (max-width: 767.9px){
.lt-md\\:mt-32{margin-top:3.2rem;}
}
@media (min-width: 768px){
.md\\:mt-32{margin-top:3.2rem;}
}
@media (min-width: 768px) and (max-width: 989.9px){
.at-md\\:mt-32{margin-top:3.2rem;}
}"
`;

exports[`variants > custom 1`] = `
"/* layer: default */
.last-child\\:mb-0>:last-child{margin-bottom:0rem;}"
`;

exports[`variants > data variant 1`] = `
"/* layer: default */
.data-\\[is-active\\]\\:s-bg-primary[data-is-active]{background-color:var(--w-s-color-background-primary);}"
`;

exports[`variants > important 1`] = `
"/* layer: default */
.\\!mb-32{margin-bottom:3.2rem !important;}
@media (min-width: 480px){
.\\!sm\\:mt-32{margin-top:3.2rem !important;}
}
@media (min-width: 768px){
.\\!md\\:-bottom-1\\/2{bottom:-50% !important;}
}"
`;

exports[`variants > negative 1`] = `
"/* layer: default */
.-inset-16{inset:-1.6rem;}
.-bottom-1\\/2{bottom:-50%;}
.-mt-32{margin-top:-3.2rem;}"
`;

exports[`variants > pseudo 1`] = `
"/* layer: default */
.focus\\:border:focus{border-width:1px;}
.part-\\[part-name\\]\\:s-bg-primary::part(part-name){background-color:var(--w-s-color-background-primary);}
.group:hover .group-hover\\:p-16{padding:1.6rem;}
.before\\:p-32::before{padding:3.2rem;}
.group\\/llama:hover .group-hover\\/llama\\:m-32{margin:3.2rem;}
.last\\:mb-0:last-child{margin-bottom:0rem;}"
`;
